[BUG] StreamReader throws ConcurrentModificationException

```
2024-07-09 15:50:31.606 +0800 ERROR [com.automq.stream.s3.S3Stream]  [streamId=4248 epoch=3] stream fetch [378728, 387067) 1048576 fail
java.util.concurrent.CompletionException: java.util.ConcurrentModificationException
        at java.base/java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:332)
        at java.base/java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:347)
        at java.base/java.util.concurrent.CompletableFuture.uniWhenComplete(CompletableFuture.java:874)
        at java.base/java.util.concurrent.CompletableFuture.uniWhenCompleteStage(CompletableFuture.java:887)
        at java.base/java.util.concurrent.CompletableFuture.whenComplete(CompletableFuture.java:2357)
        at com.automq.stream.s3.cache.blockcache.StreamReader.read(StreamReader.java:99)
        at com.automq.stream.s3.cache.blockcache.StreamReaders$Cache.lambda$read$1(StreamReaders.java:125)
        at com.automq.stream.utils.threads.EventLoop.run(EventLoop.java:49)
Caused by: java.util.ConcurrentModificationException
        at java.base/java.util.LinkedList$ListItr.checkForComodification(LinkedList.java:977)
        at java.base/java.util.LinkedList$ListItr.next(LinkedList.java:899)
        at com.automq.stream.s3.cache.blockcache.StreamReader.lambda$loadMoreBlocksWithoutData0$23(StreamReader.java:365)
        at java.base/java.util.concurrent.CompletableFuture$UniCompose.tryFire(CompletableFuture.java:1150)
        at java.base/java.util.concurrent.CompletableFuture$Completion.run(CompletableFuture.java:482)
```

--- a/metadata/src/main/java/org/apache/kafka/image/S3StreamsMetadataImage.java
+++ b/metadata/src/main/java/org/apache/kafka/image/S3StreamsMetadataImage.java
@@ -161,7 +161,6 @@ public final class S3StreamsMetadataImage extends AbstractReferenceCounted {
         int streamSetObjectIndex = 0;
         fillObjects(ctx, stream, objects, lastRangeIndex, streamObjectIndex, streamObjects, streamSetObjectIndex,
             null, null);
-        ctx.cf.complete(new InRangeObjects(streamId, objects));
     }
 
     void fillObjects(GetObjectsContext ctx, S3StreamMetadataImage stream, List<S3ObjectMetadata> objects, int lastRangeIndex,
@@ -200,6 +199,7 @@ public final class S3StreamsMetadataImage extends AbstractReferenceCounted {
                 // 1. can not find the range containing nextStartOffset, or
                 // 2. the range is the same as the last one, which means the nextStartOffset does not move on.
                 if (rangeIndex < 0 || lastRangeIndex == rangeIndex) {
+                    ctx.cf.complete(new InRangeObjects(ctx.streamId, objects));
                     break;
                 }
                 lastRangeIndex = rangeIndex;

